### PR TITLE
Fix catalog_product_entity_media_gallery not cleared when related products are being deleted

### PR DESCRIPTION
**Pull request by dkmaczynski**
### Description
Add constraint to remove `catalog_product_entity_media_gallery` rows with corresponding product pictures' paths when the product is deleted.

### Fixed Issues
1. magento/magento2#17727: `catalog_product_entity_media_gallery` not cleared when related products deleted.

#### Before applying this fix
1. Add new product,
2. add pictures to the new product,
3. check `catalog_product_entity_media_gallery` - there should be a row for each of the product pictures,
4. delete the product,
5. check the database table again - there still are rows corresponding to the product.

#### After applying this fix
1. Do steps 1-4 from the list above,
2. check the database table again - there are no rows corresponding to the product now.

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/magento2?pullRequest=18108) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dev.magento.com/magento/magento2?pullRequest=18108) before we can accept your contribution.<br/><hr/>**Dawid Mączyński** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).

**Comment by magento-engcom-team** [comment]
Hi @dkmaczynski. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me $VERSION instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

**Review by orlangur — CHANGES_REQUESTED** [review]
Nowadays all changes must be applied to `2.3-develop` first (see https://devdocs.magento.com/guides/v2.3/contributor-guide/contributing.html#rules). Please prepare a new pull request, this one should be `on hold` until the latter is merged.

**Comment by orlangur** [comment]
Closing due to inactivity. Feel free to reach me out anytime later if you wish to continue work on this pull request and it will be reopened.
Thanks for collaboration!

**Pull request closed without merging**
